File: project/src/main/java/com/example/project/entities/Employee.java
package com.example.project.entities;

import java.io.Serializable;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;


@Data
@Document(collection = "Employee")
public class Employee implements Serializable {
	@Id
	private int employee_id;
	private String name;
	private String position;
	private String contract_details;
    private Department department;
}

File: project/angular_project/src/app/models/employee.ts
import { Department } from "./department";

export class Employee {

  employee_id: number;
  name: string;
  position: string;
  contract_details: string;
  department: Department;

  constructor() {
    this.employee_id = 0;
    this.name = "";
    this.position = "";
    this.contract_details = "";
    this.department = new Department();
  }
}
